### PR TITLE
Get posts from ID/subreddit, GetParent from comments

### PR DESCRIPTION
**Pull request by jakelauer**
In this request I've added two things:
- Method to get a post by supplying the post's subreddit and linkId (rather than the full URL)
- Method on Comment to get the comment's parent Thing

### CODE REVIEW

File: RedditSharp/Reddit.cs [hunk 1]
```diff
@@ -358,25 +371,25 @@ public Thing GetThingByFullname(string fullname)
...
+		public Comment GetComment(string subreddit, string name, string linkName)
+		{
+			try
+			{
+				if (linkName.StartsWith("t3_"))
+					linkName = linkName.Substring(3);
+				if (name.StartsWith("t1_"))
+					name = name.Substring(3);
+
+				var url = string.Format("http://reddit.com" + GetCommentUrl, subreddit, linkName, name);
```
Comment (SirCmpwn) [inline RedditSharp/Reddit.cs]: Don't explicitly use "http://reddit.com". Why did you change this?

**Comment by SirCmpwn** [comment]
The indentation on this PR is wrong - use spaces instead of tabs.

Looks good aside from these two minor issues.

File: RedditSharp/Reddit.cs (hunk 1 above)
Comment (mike-eason) [inline RedditSharp/Reddit.cs]: `string.Format("http://reddit.com" + GetCommentUrl, subreddit, linkName, name);`

This is completely wrong. The string has no format items, i.e `{0}`.

Comment (SirCmpwn) [inline RedditSharp/Reddit.cs]: GetCommentUrl has format items, @mike-eason.

Comment (mike-eason) [inline RedditSharp/Reddit.cs]: Yeah, I realised my mistake a few minutes ago. I will down my pitchfork, I apologize.

Comment (jakelauer) [inline RedditSharp/Reddit.cs]: @SirCmpwn For some reason, the code errors for me without the explicit use of http://reddit.com because Uri() doesn't work without a fully qualified URL. Am I doing something wrong?

Comment (jakelauer) [inline RedditSharp/Reddit.cs]: also I can fix the tabs, for sure!

Comment (SirCmpwn) [inline RedditSharp/Reddit.cs]: Make use of the WebAgent.

**Comment by jakelauer** [comment]
I've updated the request! I am not entirely sure if I've done it the way you recommended, let me know if I should change further :)

File: RedditSharp/Reddit.cs [hunk 2]
```diff
@@ -288,6 +289,18 @@ public JToken GetToken(Uri uri)
...
+        public Post GetPost(string subreddit, string linkId)
+        {
+            if (linkId.IndexOf("_", StringComparison.Ordinal) > 0)
+            {
+                linkId = linkId.Substring(3, linkId.Length - 3);
+            }
+            var basePostUrl = string.Format(GetPostUrlBase, subreddit, linkId);
+            var url = string.Format(GetPostUrl, basePostUrl);
+            var get = WebAgent.CreateGet(url);
+            return GetPost(get.RequestUri);
```
Comment (SirCmpwn) [inline RedditSharp/Reddit.cs]: Can you refactor this to avoid calling CreateGet just to get the Uri, and CreatePost again later?

**Comment by jakelauer** [comment]
Sorry for my ignorance - is there a way to do that? From what I can see, all of the logic that actually prepends the protocol and domain to the relative URL happens in CreateRequest. I could duplicate that code somewhere else, but that seems like a waste. Perhaps I'm missing something?

**Comment by SirCmpwn** [comment]
It'd probably involve refactoring the existing implementation of GetPost as well.

**Comment by jakelauer** [comment]
Ah, gotcha. Will do!

**Comment by jakelauer** [comment]
Alright, I think I've got something that works pretty well. Mind taking a look?

**Comment by SirCmpwn** [comment]
Your indentation is still pretty screwed up. Read [the diff](https://github.com/SirCmpwn/RedditSharp/pull/219/files).

**Comment by jakelauer** [comment]
Hmm, I'm not seeing where indentation is messed up. It all looks the same to me, do you see specific lines that have problems?

**Comment by jakelauer** [comment]
Oooh, nevermind. I see the issues. Weird! Will fix.

File: RedditSharp/Reddit.cs [hunk 3]
```diff
@@ -275,25 +276,44 @@ public Domain GetDomain(string domain)
...
+			if (url.EndsWith("/"))
+				url = url.Remove(url.Length - 1);
+
+			var request = WebAgent.CreateGet(string.Format(GetPostUrl, url));
+			var response = request.GetResponse();
+			var data = WebAgent.GetResponseString(response.GetResponseStream());
+			var json = JToken.Parse(data);
+
+			return json[0]["data"]["children"].First;
+	    }
```
Comment (SirCmpwn) [inline RedditSharp/Reddit.cs]: Here.

**Comment by jakelauer** [comment]
Alright, I think I fixed it.

**Comment by SirCmpwn** [comment]
If you don't actually need a function that uses an ID and a subreddit, I'd rather not include one version with a Uri and one version with a string. A Uri version alone would be fine, since it's easy to convert a string to a Uri.

File: RedditSharp/Reddit.cs [hunk 4]
```diff
@@ -358,29 +378,34 @@ public Thing GetThingByFullname(string fullname)
...
+        {
+            try
+            {
+                if (linkName.StartsWith("t3_"))
+                    linkName = linkName.Substring(3);
+                if (name.StartsWith("t1_"))
+                    name = name.Substring(3);
+
+                var url = string.Format(GetCommentUrl, subreddit, linkName, name);
+                var fullUrl = string.Format(GetPostUrl, url);
```
Comment (CrustyJew) [inline RedditSharp/Reddit.cs]: Not sure this should be changed, and it's formatting GetPostUrl to get a full url for a comment which doesn't seem right.

File: RedditSharp/Reddit.cs [hunk 5]
```diff
@@ -275,24 +276,43 @@ public Domain GetDomain(string domain)
...
-            var data = WebAgent.GetResponseString(response.GetResponseStream());
-            var json = JToken.Parse(data);
-
-            return json[0]["data"]["children"].First;
+        public JToken GetToken(Uri uri)
+        {
+            return GetToken(uri.AbsoluteUri);
+        }
+
+        public JToken GetToken(string url)
```
Comment (CrustyJew) [inline RedditSharp/Reddit.cs]: I don't think this is necessary, better to just leave GetToken(Uri) implementation, same goes for GetPost

**Comment by CrustyJew** [comment]
The line endings aren't correct for this pull request either. If you can correct those and address the 2 comments I left, I'd appreciate it.
